### PR TITLE
Harsh Weekly volunteer summary dashboard

### PR DESCRIPTION
**Pull request by hbodgal**
# Description
This PR is base branch for the new feature being added to the app. (Feature : Total Org Summary)

## Related PRS (if any):
This frontend PR is related to the (https://github.com/OneCommunityGlobal/HGNRest/pull/994) backend PR.
…

## Main changes explained:
Created Base branch for new feature.
Added redux and connected with API.
Merged #2423 #2414.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ Reports menu → Total Org Summary
6. Check if you can open page without any errors.
7. Check if you can see multiple Accordians.

### CODE REVIEW

**Comment by angelalalacheng** [comment]
Hi hbodgal,
I switch the frontend repo to branch `weekly-volunteer-summary-dashboard` and backend repo to branch `abi-overview-report` and use the Admin account to login. 

I don't see any options called `Total Org Summary` in Report page.
Screenshot 2024-07-11 at 11 35 46 AM
Or, I guess what you mean is the following (?)
Screenshot 2024-07-11 at 11 36 23 AM


For everyone who wants to review PR, remember to `npm install react-collapsible`.

**Review by angelalalacheng — APPROVED** [review]
I can find the option on the dropdown menu. And, all the results can see on the page. There seems a minor error on api but the frontend works as you expected. Dark mode shows display good as well. So, I approved it.
Screenshot 2024-07-12 at 9 50 15 AM
Screenshot 2024-07-12 at 9 52 57 AM

**Review by jinxyou — DISMISSED** [review]
I can open the total org summary page without errors, and I can see the accordions. 
Screenshot 2024-07-13 142854

**Comment by Parth-tech** [comment]
Hi Harsh,
The PR seems great. The required functionality works fine and the codebase looks great too.

`Image Reference:`
Screenshot 2024-07-13 at 3 52 47 PM
Screenshot 2024-07-13 at 4 24 05 PM

**Review by DiyaWadhwani — COMMENTED** [review]
Hi Harsh,
I checked in to your branch but I cannot get the server to run as it is failing to compile with a file not found. I have attached a screenshot below. Kindly review this.
Thank you

Screenshot 2024-07-14 at 8 47 09 AM

**Review by SaiLavanya1 — APPROVED** [review]
Hi,
The changes introduced in this PR have been successfully tested, and the functionality works as expected. The UI is responsive and adapts well to smaller screens.Below is the video of running the PR #2394.
https://www.loom.com/share/6d3827ad6eba40d087c588730bf6fec4

**Review by JatinAgrawal94 — APPROVED** [review]
Changes work as expected. Page is visible.
Screenshot 2024-07-17 at 3 47 30 PM
Screenshot 2024-07-17 at 3 47 37 PM

**Comment by kabraambika** [comment]
Could you please link the Google Doc with the testing instructions? I wanted to know what is expected in the "Total Org Summary" and also the number of accordions that should be visible.

Also, Currently implementation is not supporting 'dark mode on" as shown below:
image

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -71,43 +77,49 @@ export function Header(props) {
   // Reports
   const canGetReports = props.hasPermission('getReports');
   const canGetWeeklySummaries = props.hasPermission('getWeeklySummaries');
+  const canGetWeeklyVolunteerSummary = props.hasPermission('getWeeklySummaries');
```
Comment (kabraambika) [inline src/components/Header/Header.jsx:80]: The permission "getWeeklySummaries" is used for both `canGetWeeklySummaries` and `canGetWeeklyVolunteerSummary`. Do we need a separate constant for the same permission?

**Review by kabraambika — CHANGES_REQUESTED** [review]
I have reviewed the frontend PR and provided comments for improvement.

**Review by SrikanthChowdary495 — COMMENTED** [review]
I have tested this PR. It is working in both dark mode and normal mode.  I can see the accordions. But there is API error in the console. 
totalorgsummary
totalorgsummarydarkmode

**Review by aaryaneil — COMMENTED** [review]
Tested the PR and the "Total Org Summary" option is visible and the accordians are visible too. Though I noted an error in the API.
Screenshot 2024-07-21 at 5 58 24 AM
Screenshot 2024-07-21 at 5 58 52 AM

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (hbodgal) [reply src/components/Header/Header.jsx:94]: we still need to configure other APIs to send permission specifically for Total Org Summary. Currently, it is just using permission for WeeklySummaries. Permission will be configured once we have a design ready.

Comment (kabraambika) [reply src/components/Header/Header.jsx:94]: Thanks for the explanation and for sharing the future plans for this.

**Review by kabraambika — CHANGES_REQUESTED** [review]
Thanks for sharing purpose of two similar const but can you address these observations[[Link](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2394#issuecomment-2239535966)] as well ?

**Review by kabraambika — APPROVED** [review]
I have reviewed the changes, and they are working as intended as shown below

image

**Review by ishan-miglani — APPROVED** [review]
Hi, I have verified the changes in this PR. I am able to see the Total Org Summary option in the dropdown and load the page and see the multiple Accordians. I do see an API error in the console on page load but I think that will be addressed with the backend PR. Approving, Thanks!
Screenshot 2024-07-27 at 12 25 33 AM
Screenshot 2024-07-27 at 12 26 21 AM
Screenshot 2024-07-27 at 12 29 22 AM

**Review by kabraambika — APPROVED** [review]
I have reviewed this PR along with backend PR 994 and it's working fine in light and dark mode.

**Review by ReinaT5678 — COMMENTED** [review]
I got this error here: 
Screenshot 2024-07-31 at 9 18 49 PM

However, when this issue was fixed, I couldn't find the "Total Org Summary" in the drop down menu.

Screenshot 2024-07-31 at 9 21 10 PM

**Comment by souravwalke** [comment]
I have reviewed this PR along with backend PR 994. I can see the Total Org Summary option in the dropdown, load the page, and see the multiple Accordions. 

https://www.loom.com/share/4dad8febfd124744b4ba568b743a33c5?sid=88d93f28-4f52-49fb-bee6-4ae639958360

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**